node/worker/worker.go

```diff
@@ -41,32 +43,118 @@ type Worker struct {
 	shardID uint32
 }
 
+// Returns a tuple where the first value is the txs sender account address,
+// the second is the throttling result enum for the transaction of interest.
+// Throttling happens based on the amount, frequency, etc.
+func (w *Worker) throttleTxs(selected types.Transactions, recentTxsStats types.RecentTxsStats, txsThrottleConfig *shardingconfig.TxsThrottleConfig, tx *types.Transaction) (common.Address, shardingconfig.TxThrottleFlag) {
+	chainID := tx.ChainID()
+	// Depending on the presence of the chain ID, sign with EIP155 or homestead
+	var s types.Signer
+	if chainID != nil {
+		s = types.NewEIP155Signer(chainID)
+	} else {
+		s = types.HomesteadSigner{}
+	}
+
+	var sender common.Address
+	msg, err := tx.AsMessage(s)
+	if err != nil {
+		utils.GetLogInstance().Error("Error when parsing tx into message",
+			"tx Id", tx.Hash().Hex(), "err", err)
+	} else {
+		sender = msg.From()
+	}
+
+	// already selected max num txs
+	if len(selected) > txsThrottleConfig.MaxNumTxsPerBlockLimit {
+		utils.GetLogInstance().Info("Throttling tx with max num txs per block limit",
+			"tx Id", tx.Hash().Hex(),
+			"MaxNumTxsPerBlockLimit", txsThrottleConfig.MaxNumTxsPerBlockLimit)
+		return sender, shardingconfig.TxUnselect
+	}
+
+	// throttle a single sender sending too many transactions in one block
+	if tx.Value().Cmp(txsThrottleConfig.MaxTxAmountLimit) > 0 {
```

Comment:
we should convert MaxTxAmountLimit to wei by multiplying 1e18

Reply:
I see it's fixed.

Reply:
this is all fixed